yt/fields/vector_operations.py

```diff
@@ -89,6 +90,37 @@ def _relative_vector(field, data):
                            units=field_units,
                            validators=validators)
 
+def create_los_field(registry, basename, field_units, 
+                     ftype='gas', slice_info=None,
+                     validators=None):
+    axis_order = registry.ds.coordinates.axis_order
+
+    field_comps = [(ftype, "%s_%s" % (basename, ax)) for ax in axis_order]
+    field_comps_rel = [(ftype, "relative_%s_%s" % (basename, ax)) 
+                       for ax in axis_order]
+
+    def _los_field(field, data):
+        axis = data.get_field_parameter("axis")
```

Comment:
Quick question, does `axis` need to be checked to see if it's a unit vector?

Reply:
It could be either a unit vector or a single integer for an axis-aligned l.o.s. Do you have a suggestion for a check?